packages-exp/auth-exp/src/core/user/additional_user_info.test.ts

```diff
@@ -0,0 +1,215 @@
+        const idResponse = idTokenResponse({ providerId: ProviderId.FACEBOOK });
+        expect(fromIdTokenResponse(idResponse)!.profile).to.be.empty;
+      });
+    });
+
+    describe('determines new-user status', () => {
+      it('for new users by token response', () => {
+        const idResponse = idTokenResponse({
+          providerId: ProviderId.FACEBOOK,
+          isNewUser: true
+        });
+        expect(fromIdTokenResponse(idResponse)!.isNewUser).to.be.true;
+      });
+
+      it('for new users by toolkit response kind', () => {
+        const idResponse = idTokenResponse({
+          providerId: ProviderId.FACEBOOK,
+          kind: IdTokenResponseKind.SignupNewUser
+        });
+        expect(fromIdTokenResponse(idResponse)!.isNewUser).to.be.true;
+      });
+
+      it('for old users', () => {
+        const idResponse = idTokenResponse({ providerId: ProviderId.FACEBOOK });
+        expect(fromIdTokenResponse(idResponse)!.isNewUser).to.be.false;
+      });
+    });
+
+    describe('creates generic AdditionalUserInfo', () => {
+      it('for custom auth', () => {
+        const idResponse = idTokenResponse({
+          providerId: ProviderId.CUSTOM,
+          rawUserInfo: rawUserInfoWithLogin
+        });
+        const {
+          isNewUser,
+          providerId,
+          username,
+          profile
+        } = fromIdTokenResponse(idResponse)!;
+        expect(isNewUser).to.be.false;
+        expect(providerId).to.be.null;
+        expect(username).to.be.null;
+        expect(profile).to.eq(profile);
+      });
+
+      it('for anonymous auth', () => {
+        const idResponse = idTokenResponse({
+          providerId: ProviderId.ANONYMOUS,
+          rawUserInfo: rawUserInfoWithLogin
+        });
+        const {
+          isNewUser,
+          providerId,
+          username,
+          profile
+        } = fromIdTokenResponse(idResponse)!;
+        expect(isNewUser).to.be.false;
+        expect(providerId).to.be.null;
+        expect(username).to.be.null;
+        expect(profile).to.eq(profile);
+      });
+
+      it('for missing provider IDs in response but not in token', () => {
+        const {
+          isNewUser,
+          providerId,
+          username,
+          profile
+        } = fromIdTokenResponse(
+          idTokenResponse({ rawUserInfo: rawUserInfoWithLogin })
+        )!;
+        expect(isNewUser).to.be.false;
+        expect(providerId).to.eq(ProviderId.FACEBOOK);
+        expect(username).to.be.null;
+        expect(profile).to.eq(profile);
+      });
+    });
+
+    describe('returns null', () => {
+      it('for missing provider IDs', () => {
+        const idResponse = idTokenResponse({});
+        expect(fromIdTokenResponse(idResponse)!).to.be.null;
+      });
+    });
+  });
+});
+
+function idTokenResponse(partial: Partial<IdTokenResponse>): IdTokenResponse {
+  return {
+    idToken: 'Parsing logic not implemented',
+    refreshToken: "Doesn't matter",
```

Comment:
nit: we're choosing to use `property-name` in kebab case for the property values in tests, recommend you update to stay consistent with the other tests we wrote, ie: this one should read `refreshToken: 'refresh-token'`